homeassistant/components/sensor/steam_online.py

```diff
@@ -75,24 +78,38 @@ def update(self):
         """Update device state."""
         try:
             self._profile = self._steamod.user.profile(self._account)
-            if self._profile.current_game[2] is None:
-                self._game = 'None'
-            else:
-                self._game = self._profile.current_game[2]
+            self._game = self._get_current_game()
             self._state = {
                 1: STATE_ONLINE,
                 2: STATE_BUSY,
                 3: STATE_AWAY,
                 4: STATE_SNOOZE,
-                5: STATE_TRADE,
-                6: STATE_PLAY,
-            }.get(self._profile.status, 'Offline')
+                5: STATE_LOOKING_TO_TRADE,
+                6: STATE_LOOKING_TO_PLAY,
+            }.get(self._profile.status, STATE_OFFLINE)
             self._name = self._profile.persona
             self._avatar = self._profile.avatar_medium
         except self._steamod.api.HTTPTimeoutError as error:
             _LOGGER.warning(error)
             self._game = self._state = self._name = self._avatar = None
 
+    def _get_current_game(self):
+        game_id = self._profile.current_game[0]
+        game_extra_info = self._profile.current_game[2]
+
+        if game_extra_info:
+            return game_extra_info
+
+        if game_id:
+            # Initialize app list before usage to benefit from internal caching
+            app_list = self._steamod.apps.app_list()
+            if game_id in app_list:
+                # The app list always returns a tuple
+                # with the game id and the game name
+                return app_list[game_id][1]
+
+        return NO_GAME
```

Comment:
I know this wasn't a new behavior, but instead of `NO_GAME` here, let's have this just return `None`, then change `device_state_attributes` to:

```python
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
        return {'game': self._game} if self._game else None
```

This will just omit the game attribute if the user isn't playing a game.

Reply:
I'll change that as well. Please note however that this would also be a breaking change since `'None'` and `None` would not be the same value, so people checking if any game is being played by a user can't rely on code such as `if game != 'None':` for instance. This is all me assuming that HASS does not automagically turn `'None'` into `None` at some point of course 😄

Reply:
That's true, this will be another breaking change. However, in the end it'll be better semantics, since users can check if the game attribute exists. (i.e. `if not states.sensor.steam_user.attributes.game:`)

Reply:
Will a user be able to check if a game exists that way if I use the code you wrote above? Since I do not know exactly how the inner cogs of HASS turns I just want to make sure. If there is loop with a `setattr` call for each of the attribute for instance, then the a programmer could not do `if not states.sensor.steam_user.attributes.game` since game would not be defined since the `device_state_attributes` property method would return None.

Would it then be better to leave the old code `return {'game': self._game}` and just set `self._game` to `None` instead of `'None'`.

Again, these are new things to me so I just want to make sure :)

Reply:
Good question! The way it works is that states in hass have an attribute dict attached to them. When looking up a state, this dict will always exist, but the key 'game' might not. ([This is where this happens](https://github.com/home-assistant/home-assistant/blob/8742ce035a36b3a4a176123b620fe82e55d5a4e2/homeassistant/helpers/entity.py#L218-L220)).

The code above was just pseudo-code, but the idea will be that users can check for the existence of that key.

Reply:
Alright, so the dict will always exist but the `game` key will not right? Wouldn't it be nicer if the programmer always knew that the property `game` would exist and just be able to do `if not states.sensor.steam_user.attributes.game` instead of `if getattr(states.sensor.steam_user.attributes, 'game', None)`? Or is there some "magic" somewhere that will always return False/None if a attribute is not set?

Reply:
I'm sorry if I'm wasting your time, just want to get it right the first time so that if I contribute more I don't need to ask again 😄

Reply:
It's no problem. In templates, you can do it just as was written above (had to check):
`if states.sesnor.steam_user.attributes.game`. Users would normally only be checking this in templates. In python you'd have to get the state object anyway, then you could check 
```python
if 'game' in state.attributes:
    print(state.attributes['game'])
```
or
```python
print(state.attributes.get('game', 'Default Fallback String (can be defined per usage)'))
```